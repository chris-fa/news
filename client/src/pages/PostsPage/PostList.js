import React from 'react';

const PostList = ({ posts }) => (
  <div>
    { posts.map(post =>
      <div key={ post.id } className="box">
        <h4 className="title is-4">{ post.title }</h4>
        <h6 className="subtitle is-6">{ `By ${post.author} at
          ${post.updatedAt.toLocaleDateString('en-US')}` }</h6>

        <figure className="image is-4by3">
          <img src={ post.image } alt={ post.title }/>
        </figure>

        <p>{ post.body }</p>
      </div>
    )}
  </div>
);

export default PostList;
