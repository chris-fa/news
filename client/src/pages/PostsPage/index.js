export { PostsPage } from './PostsPage';
export { default } from './PostsPage';
