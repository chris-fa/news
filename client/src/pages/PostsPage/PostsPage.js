import React from 'react';
import { connect } from 'react-redux';

import PostList from './PostList';

export class PostsPage extends React.Component {
  render() {
    const { posts } = this.props;

    return (
      <div className="container">
        <h1 className="title is-1">Posts</h1>
        <PostList posts={ posts } />
      </div>
    );
  }
};

export const mapStateToProps = state => ({
});

export const mapDispatchToProps = dispatch => ({
});

export default connect()(PostsPage);
