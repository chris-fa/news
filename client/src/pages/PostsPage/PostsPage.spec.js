import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PostsPage } from './PostsPage';
import PostList from './PostList';

describe('<PostsPage />', () => {
  const defaultProps = {
    isLoading: false,
    errorMessage: null,
    posts: [],
  };

  it('renders a <PostList>', () => {
    const props = {
      ...defaultProps,
      posts: [{}],
    };

    const wrapper = shallow(<PostsPage { ...props } />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(PostList)).toExist();
  });
});
