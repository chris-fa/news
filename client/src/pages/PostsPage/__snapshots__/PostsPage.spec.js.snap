// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsPage /> renders a <PostList> 1`] = `
<div
  className="container"
>
  <h1
    className="title is-1"
  >
    Posts
  </h1>
  <PostList
    posts={
      Array [
        Object {},
      ]
    }
  />
</div>
`;
